src/core/expression/qgsexpression.cpp

```diff
@@ -1061,6 +1061,11 @@ QString QgsExpression::formatPreviewString( const QVariant &value, const bool ht
     const QgsCoordinateReferenceSystem crs = value.value<QgsCoordinateReferenceSystem>();
     return startToken + tr( "crs: %1" ).arg( crs.userFriendlyIdentifier() ) + endToken;
   }
+  else if ( value.userType() == qMetaTypeId< QTimeZone>() )
+  {
+    const QTimeZone tz = value.value<QTimeZone>();
+    return startToken + tr( "time zone: %1" ).arg( tz.displayName( QTimeZone::GenericTime, QTimeZone::ShortName ) ) + endToken;
```

Comment:
Should we identify !QTimeZone.isValid() as "time zone: invalid" here?

Reply:
Ok, done